web-gateway/src/main/java/controllers/ItemController.java

```diff
@@ -166,65 +167,86 @@ private ItemData fromForm(ItemForm itemForm) {
             }
         });
     }
-
     public CompletionStage<Result> getItem(String itemId) {
         return doGetItem(ctx(), itemId, formFactory.form(BidForm.class));
     }
 
     private CompletionStage<Result> doGetItem(Http.Context ctx, String itemId, Form<BidForm> bidForm) {
         return requireUser(ctx, user -> loadNav(user).thenComposeAsync(nav -> {
+
             UUID itemUuid = UUID.fromString(itemId);
             CompletionStage<Item> itemFuture = itemService.getItem(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
             CompletionStage<PSequence<Bid>> bidHistoryFuture = bidService.getBids(itemUuid)
                     .handleRequestHeader(authenticate(user)).invoke();
-            return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
 
-                if (item.getStatus() == ItemStatus.CREATED && !item.getCreator().equals(user)) {
-                    return forbidden();
-                }
+                return itemFuture.thenCombineAsync(bidHistoryFuture, (item, bidHistory) -> {
+
+                    User seller = null;
+                    Optional<User> winner = Optional.empty();
+
+                    try {
+                        seller = userService.getUser(item.getCreator()).invoke().toCompletableFuture().get();
```

Comment:
I've been putting some thought on these two calls (load seller and winner) and I think that for the moment we can extract each of them into a separate blocking method.

Java (the language and the API) are not great for composition of futures (`CompletionStage`) and even less when the call depends on the result of a previous `CompletionStage` (like in this case where the calls to `UserService` depend on `item`).

So, for the time being I think we can:
1) add an issue in https://github.com/lagom/online-auction-java stating this blocking calls should be removed
2)extract these two blocks into separate private methods that return `User` or `null` (similar to what we had before) 
3) Add a `TODO` in this code stating: "`CompletableFuture.get()` is a blocking call, this code should be reviewed with CompletionStage composition."

There are further issues with the code as is: if `UserService` is down, then these two call will fail, we should work on a fallback. It's probably a good idea to enrich `Item` with one or two fields with the `seller`'s and `winner`s usernames. This way, when these calls to `UserService` fail and we can't display the name and surname, we could fallback to username. I would also make this on a separate PR.